### ISSUE TITLE
sos! method for long-time backend api call

### ISSUE THREAD

**Issue by ltm920716** [reflex-dev/reflex#1696]
**Describe the bug**
Hello，I meet the situation that I will call a long-time backend logical method such like chatgpt. 
If I use State.event-handler， the fronted UI will be blocked， even use async and await.
If I use a  separate backend api served by Fastapi and call by aiohttp，the fronted UI is blocked too.

So I want to know that in Reflex, is there any solution for situation above? Now I can not develop further ，it is a very sad thing

```
async def make_async_request(url):
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            return await response.text()


class State(rx.State):
    """The app state."""

    count: int = 0
    progress: int = 0

    # UI will be blocked by button on_click here
    async def run(self):
        res = await make_async_request('http://127.0.0.1:8003/test/12')
        self.count = res
```

**Expected behavior**
A async solution to call long-time backend api (or long-time State.event)，thanks

**Comment by masenf** [comment]
Coming soon ™️ https://github.com/reflex-dev/reflex/pull/1676

**Comment by ltm920716** [comment]
> Coming soon ™️ #1676

Hi @masenf，
That's really great，reflex dose need to implement this truly asynchronous tasks to be the useful framework！

**Comment by masenf** [comment]
@ltm920716 if you have any comments on the design / API: https://github.com/orgs/reflex-dev/discussions/1838

**Comment by masenf** [comment]
@rx.background decorator and background task api was released in 0.2.8
